examples/program_generator/src/lib.rs

```diff
@@ -0,0 +1,27 @@
+#![no_std]
+
+use codec::alloc::vec::Vec;
```

Comment:
You can use the original `alloc` instead of a reexport.

```suggestion
extern crate alloc;

use alloc::vec::Vec;
```